### PR TITLE
Another Shared Element Demo for Playground

### PR DESCRIPTION
**Pull request by mrousavy**
I've added another use-case for Shared Element Transitions to the Playground app. The main differences to the cocktail one are;

* Full-Screen Details Screen (no statusbar, no titlebar, no bottom tabs)
* Spring animation (added prototype iOS native code for now, android not yet implemented!)
* More physical feeling through scale button press feedback, spring animation, etc

Currently there are some issues with the SET implementation, which are nicely visible in this demo.

1. Images don't really work (different aspect ratios) and get glitchy
2. There are problems with the StatusBar and TitleBar
3. There are problems with the bottom tabs animation

All of these are known issues and have separate GitHub issues created.

Also, I wanted to add a drag down animation to dismiss the details screen using gesture handler and reanimated, but I couldn't get the libraries installed. Will look into this.

### CODE REVIEW

File: playground/src/screens/sharedElementTransitionAlt/PostDetailsScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+import { PostItem } from '../../assets/posts';
+
+interface Props extends NavigationComponentProps {
+  post: PostItem;
+}
+
+const PostDetailsScreen: NavigationFunctionComponent = ({
+  post,
+  componentId,
+}: Props): JSX.Element => {
```
Comment (jinshin1013) [inline playground/src/screens/sharedElementTransitionAlt/PostDetailsScreen.tsx:25]:
```suggestion
interface Props {
  post: PostItem;
}

const PostDetailsScreen: NavigationFunctionComponent<Props> = ({
  post,
  componentId,
}) => {
```

File: playground/src/screens/sharedElementTransitionAlt/PostCard.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import { Image, View, StyleSheet, Text, Dimensions, ViewProps } from 'react-native';
+import { PostItem } from '../../assets/posts';
+
+export default function PostCard({
+  post,
+  style,
+  ...passThroughProps
+}: { post: PostItem } & ViewProps): JSX.Element {
```
Comment (jinshin1013) [inline playground/src/screens/sharedElementTransitionAlt/PostCard.tsx:9]:
```suggestion
type PostCardProps = {
  post: PostItem
} & ViewProps

export default function PostCard({
  post,
  style,
  ...passThroughProps
}: PostCardProps) {
```

**Review by jinshin1013 — CHANGES_REQUESTED** [review]
Great that you are adding more examples of SET animation 🔥 
I haven't tried it myself yet but just some minor code style changes to keep the code consistent 🤓

**Comment by mrousavy** [comment]
@jinshin1013 facts, makes way more sense, thanks!

**Comment by jinshin1013** [comment]
🤔 For some reason, I was unable to run this example on my local setup saying these screens aren't registered. I'll try again tonight. @mrousavy Sorry for the delay 🙏

**Comment by mrousavy** [comment]
@jinshin1013 that's weird, I cloned a clean version of my branch and everything ran fine 🤔

**Comment by jinshin1013** [comment]
@mrousavy I've found that `react-native-reanimated` is not properly installed on iOS causing the error I was seeing. I've created a PR which fixes some native modules not being installed on iOS https://github.com/wix/react-native-navigation/pull/6537. Otherwise the animation looks great despite some of the issues you've pointed out!

**Review by jinshin1013 — APPROVED** [review]
Again sorry for the delay in reviewing the PR :( This is an amazing stepping stone to showcase SET animations. Great work!

**Comment by mrousavy** [comment]
@jinshin1013 awesome! We should watch out for the spring animation I've added though, since I also added `'spring'` to the interpolation types (which are exported to the public API), so I'd suggest we remove that for now and leave it a "hidden functionality" until https://github.com/wix/react-native-navigation/issues/6431 is implemented.

**Comment by mrousavy** [comment]
I've updated some stuff:

1. merged master into the branch
2. Used reanimated and gesture-handler to create a cool dismiss gesture by dragging down on the image (only enabled on iOS)
3. Added blurviews to spice up the look of the post cards

Looks a lot cleaner now, but there are a lot of things not working right now.

Stuff like #6519, #6510, #6505, #6391 and more can be reproduced using this new shared element transitions demo.

**Comment by guyca** [comment]
> I've updated some stuff:
> 
> 1. merged master into the branch
> 2. Used reanimated and gesture-handler to create a cool dismiss gesture by dragging down on the image (only enabled on iOS)
> 3. Added blurviews to spice up the look of the post cards
> 
> Looks a lot cleaner now, but there are a lot of things not working right now.
> 
> Stuff like #6519, #6510, #6505, #6391 and more can be reproduced using this new shared element transitions demo.

YES!!! This is so amazing!

**Comment by mrousavy** [comment]
[Here's](https://github.com/wix/react-native-navigation/blob/34be383db827250a2e086755a12bdb7a1eb9d8ce/playground/src/screens/sharedElementTransitionAlt/PostsListScreen.tsx#L11-L20) a list of things that aren't working which this demo should demonstrate. This demo was mainly built for iOS, but should work fine for Android as well.

**Review by yogevbd — APPROVED** [review]

**Pull request merged**